ee/clickhouse/views/test/test_clickhouse_funnel_person.py

```diff
@@ -187,22 +187,7 @@ def test_basic_pagination_with_deleted(self, delete_person_patch):
         people = j["results"][0]["people"]
         next = j["next"]
         self.assertEqual(0, len(people))
-        self.assertNotEqual(None, next)
-
-        response = self.client.get(next)
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        j = response.json()
-        people = j["results"][0]["people"]
-        next = j["next"]
-        self.assertEqual(0, len(people))
-        self.assertNotEqual(None, next)
-
-        response = self.client.get(next)
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        j = response.json()
-        people = j["results"][0]["people"]
-        self.assertEqual(0, len(people))
-        self.assertEqual(None, j["next"])
+        self.assertIsNone(next)
```

Comment:
@EDsCODE was this test previously correct? It seems like not - my changes fixed that.

Reply:
Wait what what what? How was this always passing 👀 

because https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/funnels/funnel_persons.py#L21

In this sense, makes sense 🤔 - but feels wrong lol

Reply:
Please check my assumptions here, @macobo. I think this makes sense now because: Changing engines changes how this query behaves.

Earlier, our Postgres<->Clickhouse persons deletes were out of sync, so CH would return persons that have been deleted (still unsure how exactly this happened, since we had the `max(is_deleted)` check in our queries). These were discarded by the `PeopleSerializer` and returned a set of size < 100. If we use the final size to determine pagination, we're in a bind, since these deleted peeps have been removed, so we used the original result set.

Now, Postgres<-> Clickhouse person deletes are in sync, so CH would not return persons that have been deleted (this should've always been the case? But maybe I'm missing something from the big discussion on distinct_ids-person_ids issues). So, we get the correct value the first time around.

Reply:
Leaving the discussion about the test aside, and just replying to this:

> still unsure how exactly this happened, since we had the max(is_deleted) check in our queries

The query was fine, it's just that the replacing merge tree was (under certain racy conditions) getting rid of rows where `is_deleted` was 1. So where `person_distinct_id` should have looked something like this:

| distinct_id | person_id | is_deleted |
| :--------: | :--------: | :--------: |
| id1 | Person A | 1 |
| id1 | Person B | 1 |
| id1 | Person C | 0 |

It would actually look like this:

| distinct_id | person_id | is_deleted |
| :--------: | :--------: | :--------: |
| id1 | Person A | 0 |
| id1 | Person B | 0 |
| id1 | Person C | 0 |

This is because we should have gotten rid of these rows:

| distinct_id | person_id | is_deleted |
| :--------: | :--------: | :--------: |
| id1 | Person A | 0 |
| id1 | Person B | 0 |

But instead it got rid of these:

| distinct_id | person_id | is_deleted |
| :--------: | :--------: | :--------: |
| id1 | Person A | 1 |
| id1 | Person B | 1 |

Reply:
Ahhhh, right, that makes sense! Thanks!

New problem: I've never seen this test flake, so I'm still a bit unsure about how this test reliably reproduces the out-of-syncness and tests that behaviour. (or used to, until this PR)

Reply:
(Don't let me block any approvals or anything - just a bit curious about this. It seems very surprising!)